samples/Azure.Management.Storage/Generated/StorageModelFactory.cs

```diff
@@ -0,0 +1,958 @@
+        {
+            return new DeleteRetentionPolicy(enabled, days);
+        }
+
+        /// <summary> Initializes a new instance of ChangeFeed. </summary>
+        /// <param name="enabled"> Indicates whether change feed event logging is enabled for the Blob service. </param>
+        /// <param name="retentionInDays"> Indicates the duration of changeFeed retention in days. Minimum value is 1 day and maximum value is 146000 days (400 years). A null value indicates an infinite retention of the change feed. </param>
+        /// <returns> A new <see cref="Models.ChangeFeed"/> instance for mocking. </returns>
+        public static ChangeFeed ChangeFeed(bool? enabled = null, int? retentionInDays = null)
+        {
+            return new ChangeFeed(enabled, retentionInDays);
+        }
+
+        /// <summary> Initializes a new instance of RestorePolicyProperties. </summary>
+        /// <param name="enabled"> Blob restore is enabled if set to true. </param>
+        /// <param name="days"> how long this blob can be restored. It should be great than zero and less than DeleteRetentionPolicy.days. </param>
+        /// <param name="lastEnabledOn"> Deprecated in favor of minRestoreTime property. </param>
+        /// <param name="minRestoreOn"> Returns the minimum date and time that the restore can be started. </param>
+        /// <returns> A new <see cref="Models.RestorePolicyProperties"/> instance for mocking. </returns>
+        public static RestorePolicyProperties RestorePolicyProperties(bool enabled = default, int? days = null, DateTimeOffset? lastEnabledOn = null, DateTimeOffset? minRestoreOn = null)
+        {
+            return new RestorePolicyProperties(enabled, days, lastEnabledOn, minRestoreOn);
+        }
+
+        /// <summary> Initializes a new instance of LastAccessTimeTrackingPolicy. </summary>
+        /// <param name="enable"> When set to true last access time based tracking is enabled. </param>
+        /// <param name="name"> Name of the policy. The valid value is AccessTimeTracking. This field is currently read only. </param>
+        /// <param name="trackingGranularityInDays"> The field specifies blob object tracking granularity in days, typically how often the blob object should be tracked.This field is currently read only with value as 1. </param>
+        /// <param name="blobType"> An array of predefined supported blob types. Only blockBlob is the supported value. This field is currently read only. </param>
+        /// <returns> A new <see cref="Models.LastAccessTimeTrackingPolicy"/> instance for mocking. </returns>
+        public static LastAccessTimeTrackingPolicy LastAccessTimeTrackingPolicy(bool enable = default, Name? name = null, int? trackingGranularityInDays = null, IEnumerable<string> blobType = null)
+        {
+            blobType ??= new List<string>();
+
+            return new LastAccessTimeTrackingPolicy(enable, name, trackingGranularityInDays, blobType?.ToList());
```

Comment:
nit:  I don't think we need the blotType?. here since we are guaranteed its not null with [line 82](https://github.com/Azure/autorest.csharp/pull/2947/files#diff-12b4e2c93b7f384212c026136870359882fb7d7f7ac027b7d49bc2146a5f897cR82).

Reply:
well this comes from our common code writer extension methods - it checks if the instance is nullable, if so it will append `?.ToList` otherwise `.ToList`. I think this is kind of redundant because a thing that supports `ToList` is always nullable.
But I think to change this might involve a lot of unexpected things, I would like to leave it as is because it works.